DocumentService/Repositories/IDocumentRepository.cs

```diff
@@ -42,13 +43,13 @@ public interface IDocumentRepository
         /// </summary>
         /// <param name="documentInfo">Document object with updated data</param>
         /// <returns>True if successful</returns>
-        Task<bool> Update(DocumentInfo documentInfo);
+        Task<bool> Update(DocumentDTO documentDTO, Guid id, HttpContext httpContext);
 
         /// <summary>
         /// Updates the database with a new document
         /// </summary>
-        /// <param name="documentInfo">DocumentInfo object created in the controller</param>
+        /// <param name="documentDTO">Document DTO object created in the controller to populate the DocumentInfo entity</param>
         /// <returns>Number of entities added</returns>
-        Task<int> UploadDocumentAsync(DocumentInfo documentInfo);
+        Task<int> UploadDocumentAsync(DocumentDTO documentDTO, HttpContext httpContext);
```

Comment:
Same as the above suggestion the HttpContext should not be used in the DocumentRepository.
```suggestion
        Task<int> UploadDocumentAsync(DocumentDTO documentDTO);
```